File: algorithm/3. Longest Substring Without Repeating Characters/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
    	int max=0,lastbit=-1;
        unordered_map<char,int> m;
        for (int i =0; i < s.length();i++)
        {
        	if(m.find(s[i])!=m.end() && lastbit < m[s[i]])
        	{
        		lastbit = m[s[i]];
        	}
        	if ( i - lastbit > max )
        		max = i - lastbit;
        	m[s[i]]=i;
        }
        return max;
    }
};

File: algorithm/3. Longest Substring Without Repeating Characters/3.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        map={}
        lastbit, max =-1 , 0
        for i in xrange(len(s)):
        	if map.has_key(s[i])== True and lastbit < map[s[i]]:
        		lastbit=map[s[i]]
        	if i - lastbit > max:
        		max = i - lastbit
        	map[s[i]]=i

        return max